langchain4j-open-ai/src/main/java/dev/langchain4j/model/openai/OpenAiTokenizer.java

```diff
@@ -72,6 +64,23 @@ public OpenAiTokenizer(String modelName) {
         this.encoding = Encodings.newLazyEncodingRegistry().getEncodingForModel(modelName);
     }
 
+    public static Tokenizer getTokenizerOrDefault(Tokenizer specifiedTokenizer, String modelName) {
+        if (specifiedTokenizer != null) {
+            // Use the tokenizer provided by the consumer, if present
+            return specifiedTokenizer;
+        }
+        if (modelName != null && !modelName.trim().isEmpty()) {
+            // Attempt to retrieve the model's associated tokenizer
+            OpenAiTokenizer candidate = new OpenAiTokenizer(modelName);
+            if (candidate.encoding.isPresent()) {
+                return candidate;
+            }
+        }
+        // Failing the other options, default to GPT 3.5's tokenizer, since it is the most prevalent, and is used by
+        // other (non-OpenAI) models
+        return new OpenAiTokenizer(GPT_3_5_TURBO);
```

Comment:
```suggestion
        return new OpenAiTokenizer(GPT_4_O_MINI);
```
I guess GPT_4_O_MINI is most commonly used right now, 3.5 is deprecated

Reply:
On a second thought, perharps we should return `null` here instead of a tokenizer with a default model name?
Since `Tokenizer` is not required for basic functionality of `(Streaming)ChatLanguageModel` any more, it is ok to return `null`. If user wants to estimate token count and is using some LLM provider/name that is not supported by tiktoken, perharps it is better to get NPE and configure `Tokenizer` explicitly than get wrong token count...

WDYT?